src/js/store/store.js

```diff
@@ -3,26 +3,34 @@ import {
     applyMiddleware,
     compose
 } from 'redux';
+import {createLogger} from 'redux-logger';
 import {createBrowserHistory} from 'history';
 import {routerMiddleware} from 'connected-react-router';
 import thunkMiddleware from 'redux-thunk';
 import promiseMiddleware from 'redux-promise-middleware';
 import reducers from './reducers';
+import api from '../api.js';
 
 const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // eslint-disable-line no-underscore-dangle
 const history = createBrowserHistory();
 let store;
 
+const middleware = [
+    routerMiddleware(history),
+    thunkMiddleware.withExtraArgument(api),
```

Comment:
if we are going to use `thunk`, might as well give it an extra argument like `api` cause it's very convenient